### PR TITLE
TE-6915 using new facade method to check table existance

### PR DESCRIPTION
**Pull request by gechetspr**
- Developer(s): @gechetspr

- Ticket: https://spryker.atlassian.net/browse/TE-6915

- Release Group: https://release.spryker.com/release-groups/view/3123

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   EventBehavior         | minor                 |  Propel                     |

-----------------------------------------

##### Change log

Improvements

- Added `PropelFacadeInterface` to dependencies.
- Adjusted `TriggerManager` in order to use new table existence check instead of try catch block.

Deprecations

- Deprecated `EventBehaviorQueryNotExistsException`.
- Deprecated `EventBehaviorQueryContainer::checkRequiredClasses()`.

### CODE REVIEW

**Review by bezpiatovs — APPROVED** [review]

**Review by stereomon — APPROVED** [review]

**Comment by dereuromark** [comment]
Can you rebase this on top of master, so that GitHubActions is used?
And we can see those results then in Release App for validation (green)?

File: src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php [hunk 1]
```diff
@@ -80,14 +90,20 @@ public function triggerRuntimeEvents()
...
         try {
-            $events = $this->queryContainer->queryEntityChange($processId)->find()->getData();
-            static::$eventBehaviorTableExists = true;
-        } catch (PropelException | ConnectionException | EventBehaviorQueryNotExistsException $e) {
+            if(!$this->eventBehaviorTableExists()){
+                static::$eventBehaviorTableExists = false;
+
+                return;
+            }
+        } catch ( \Throwable $exception) {
```
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php:98]: We shouldn't silently ignore exceptions, especially if we no longer expect any (We do the check for table/query existence beforehand).
Please leave only `ConnectionException` here.

File: src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php [hunk 2]
```diff
@@ -80,14 +90,20 @@ public function triggerRuntimeEvents()
 
         $processId = RequestIdentifier::getRequestId();
         try {
-            $events = $this->queryContainer->queryEntityChange($processId)->find()->getData();
-            static::$eventBehaviorTableExists = true;
-        } catch (PropelException | ConnectionException | EventBehaviorQueryNotExistsException $e) {
```
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php:85]: `EventBehaviorQueryNotExistsException` no longer used, and can be deprecated.
1. `Throw` annotation in the query container should be deprecated. 
2. Functionality that throws an exception in query container should be moved to protected method and deprecated.
3. The exception class should be deprecated.

File: src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php [hunk 3]
```diff
@@ -10,18 +10,20 @@
...
 use Spryker\Zed\EventBehavior\Dependency\Service\EventBehaviorToUtilEncodingInterface;
 use Spryker\Zed\EventBehavior\EventBehaviorConfig;
 use Spryker\Zed\EventBehavior\Persistence\EventBehaviorQueryContainerInterface;
-use Spryker\Zed\EventBehavior\Persistence\Exception\EventBehaviorQueryNotExistsException;
 use Spryker\Zed\EventBehavior\Persistence\Propel\Behavior\EventBehavior;
 use Spryker\Zed\Kernel\RequestIdentifier;
 
 class TriggerManager implements TriggerManagerInterface
 {
+    protected const TABLE_NAME_EVENT_BEHAVIOR_ENTITY_CHANGE = 'spy_event_behavior_entity_change';
```
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php:25]: Please add `@uses` link to \Orm\Zed\EventBehavior\Persistence\Map\SpyEventBehaviorEntityChangeTableMap::TABLE_NAME

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
Instead of `Adjusted TriggerManager in order to use new table existence check instead of try catch block.` in changelogs please mention the facade method that impacted.

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php [hunk 4]
```diff
@@ -112,4 +107,19 @@ public function eventBehaviorTableExists()
             return false;
         }
     }
+
+    /**
+     * @deprecated Will be removed without replacement.
+     *
+     * @throws EventBehaviorQueryNotExistsException
+     */
+    protected function checkRequiredClasses():void
```
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:116]:
```suggestion
    protected function checkRequiredClasses(): void
```

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php [hunk 5]
```diff
@@ -112,4 +107,19 @@ public function eventBehaviorTableExists()
             return false;
         }
     }
+
+    /**
+     * @deprecated Will be removed without replacement.
+     *
+     * @throws EventBehaviorQueryNotExistsException
```
Comment (asmarovydlo) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:114]:
```suggestion
     * @throws EventBehaviorQueryNotExistsException
     * 
     * @return void

```

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
Deprecations should be mentioned in the changelogs.

**Review by asmarovydlo — APPROVED** [review]

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php (hunk 5 above)
Comment (gechetspr) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:114]: Please use FQCN in docblocks

**Review by gechetspr — COMMENTED** [review]
Small change required.

Also I can't approve this PR as long as I want the one who created it
@asmarovydlo

**Review by asmarovydlo — APPROVED** [review]

File: src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php [hunk 6]
```diff
@@ -112,4 +107,19 @@ public function eventBehaviorTableExists()
             return false;
         }
     }
+
+    /**
+     * @deprecated Will be removed without replacement.
+     *
+     * @throws \Spryker\Zed\EventBehavior\Persistence\Exception\EventBehaviorQueryNotExistsException
+     */
+    protected function checkRequiredClasses(): void
```
Comment (gechetspr) [inline src/Spryker/Zed/EventBehavior/Persistence/EventBehaviorQueryContainer.php:116]: This is very generic name for the query container. I would prefer smth like `eventBehaviorEntityClassExists`

**Review by gechetspr — COMMENTED** [review]
Can't approve because I was created PR, but all good from my side

File: composer.json [hunk 7]
```diff
@@ -9,6 +9,7 @@
     "spryker/event-dispatcher-extension": "^1.0.0",
     "spryker/kernel": "^3.49.0",
     "spryker/laminas": "^1.0.0",
+    "spryker/propel": "^3.28.0",
```
Comment (dereuromark) [inline composer.json:12]: For tests we need to use "dev-... as" here for them to run
Did you do that before? So we know all is good when releasing?

I wonder if we need to fix up the auto release process to include these aliasing changes for release.
So we have better testing beforehand.

**Pull request merged**